python_ta/contracts/__init__.py

```diff
@@ -658,6 +658,8 @@ def parse_assertions(obj: Any, parse_token: str = "Precondition") -> list[str]:
         for line in lines[first + 1 :]:
             if line.startswith("-"):
                 assertion = line[1:].strip()
+                # Strip comments from line
+                assertion = assertion.split("#")[0]
```

Comment:
You're on the right track here, but this is too aggressive a change to make. You can't assume that every `#` delineates a comment (consider, for example, a string literal that contains `"#"`).

There's still something weird going on that is the root of the problem: where are the extra parentheses coming? I suggest looking for where the function `parse_assertions` (or something similar) is being used in the control flow graph visitor.